Enables bitbucket cloud provider in controller

sm43 opened the pull request:
this enables bitbucket cloud integration with controller
using webhook.

Signed-off-by: Shivam Mukhade <smukhade@redhat.com>

<!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

# Changes

<!-- Describe your changes here- ideally you can get that description straight from
your descriptive commit message(s)! -->

# Submitter Checklist

- [x] ♽  Run `make test lint` before submitting a PR (ie: via [pre-push github hook](../hack/dev/prep-push-hook) no need to waste CPU cycle on CI 
- [ ] 📖 If you are adding a user facing feature or make a change of the behavior, please make sure to document it
- [ ] 🧪 100% coverage is not a target but most of the time we would rather have a unit test if you make a code change.
- [ ] 🎁 If that's something that is possible to do please make sure to check if we can add a e2e test.
- [ ] 🔎 If there is a flakiness in the CI tests then make sure to get the flakyness fixed before merging or if that's not possible there is a good reason to bypass it.

_See [the developer guide](https://github.com/openshift-pipelines/pipelines-as-code/blob/main/docs/development.md) for a bit more details._

Commit:
Enables bitbucket cloud provider in controller

this enables bitbucket cloud integration with controller
using webhook.

Signed-off-by: Shivam Mukhade <smukhade@redhat.com>

chmouel commented on pkg/provider/bitbucketcloud/events.go:
```diff
@@ -100,19 +116,32 @@ func (v *Provider) ParsePayload(ctx context.Context, run *params.Run, request *h
-		return nil, fmt.Errorf("payload is not coming from the public bitbucket cloud ips as defined here: %s",
-			bitbucketCloudIPrangesList)
-	}
+	// TODO: figure this out!
```
Arf we reallly need to figure this out now before merging :D 

Without this anyone can use any openshift/kubernetes cluster to do anything they want.

sm43 replied:
😄 ah ok ! let me look into it..

sm43 replied:
updated! ptal :)

Commit:
bitbucket cloud: enables source ip check

this enables the source ip check for bitbucket cloud which
can be configured through pac configmap.

Signed-off-by: Shivam Mukhade <smukhade@redhat.com>

chmouel commented:
/retest

chmouel commented:
SG!

let's wait the infra is back before merging tho

sm43 commented:
@chmouel this is failing because we are getting source ip from header using `X-Forwarded-For`
but the request is forwarded from smee so ip is not the orignal one. `X-Original-Forwarded-For` would give the orignal IP
but I am not sure without using smee whether `X-Original-Forwarded-For` field will be there in header or not?

chmouel commented:
@sm43 it is disabled in e2e tests for that reason: 

https://github.com/openshift-pipelines/pipelines-as-code/blob/main/.github/workflows/kind.yaml#L158

i guess it's failing otherwise?

sm43 commented:
> @sm43 it is disabled in e2e tests for that reason:

then it should pass I ran it locally disabling it, it passed

chmouel commented:
there is something else then that needs  fixing for e2e to pass 🤔 

maybe look at the logs of the pods of the e2e to see if there is any hints,

chmouel commented:
i guess you ran the unittest/lint and alll manually?

sm43 commented:
> 

yep the git hook did ! 😄
